### PR TITLE
Add notes on ROS2 approaches and add actions

### PR DESCRIPTION
**Pull request by BorjaOuterelo**
Expands Robotis projects approach regarding MW and protocol used.
Adds MW actions related to current OFERA work.

### CODE REVIEW

**Review by richiware — DISMISSED** [review]

**Review by ralph-lange — DISMISSED** [review]

File: articles/embedded.md [hunk 1]
```diff
@@ -69,15 +69,21 @@ To answer these questions, both the OFERA EU project as well as several others h
 | A-BUILD-ARDUINO | Q-BUILD | Arduino Build | Robotis has explored building all the libraries using the Arduino IDE. This required some manual changes and thus does not scale, but can get you off the ground. | TODO |
 | A-NUTTX-LIBCXX | Q-LANG | C++11/higher support for NuttX | Build libxx from the LLVM project on NuttX, as a pre-requisite to building rclcpp. | TODO |
 | A-PERF-RCLCPP-RESOURCE | Q-PERF | Determine resource use of rclcpp | |  TODO |
-
+| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have proved that it is a suitable protocol to seamless communicate microcontrollers with ROS 2. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```
Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

Comment (iluetkeb) [reply articles/embedded.md]: I found "proved", with its connotation of formality a bit too strong. "demonstrate" should suffice.

File: articles/embedded.md [hunk 2]
```diff
@@ -69,15 +69,21 @@ To answer these questions, both the OFERA EU project as well as several others h
...
+*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [http://micro-ros.com](http://micro-ros.com).
```
Comment (iluetkeb) [inline articles/embedded.md]: Have you guys reached a consensus on the use of the domains with Erle? Personally, I would prefer linking to the Github-page for micro-ROS only.

Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [https://microros.github.io/micro-ROS/](https://microros.github.io/micro-ROS/).
```
```suggestion
*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [http://micro-ros.com](http://micro-ros.com).
```

Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [https://microros.github.io/micro-ROS/](https://microros.github.io/micro-ROS/).
```

File: articles/embedded.md [hunk 3]
```diff
@@ -69,15 +69,21 @@ To answer these questions, both the OFERA EU project as well as several others h
...
 
-*   [ROS 2 library for OpenCR by ROBOTIS](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ROS2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2)
+*   [ROS 2 library for OpenCR by ROBOTIS](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ros2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2):
+  Project enabling communication between Robotis' OpenCR board and ROS 2 nodes as part of the TurtleBot3 platform. This communication is enabled using DDS-XRCE implementation, Micro XRCE-DDS.
```
Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
  Project enabling communication between Robotis' OpenCR board and ROS 2 nodes as part of the TurtleBot3 platform. This communication is enabled using a DDS-XRCE implementation, Micro XRCE-DDS.
```

**Review by iluetkeb — CHANGES_REQUESTED** [review]
I like the additions. A few grammatical changes are suggested.

One note that I would like a comment on is the use of the micro-ros.com domain. I thought we had decided to use the Github IO page only. I may be wrong, of course, please clarify.

File: articles/embedded.md (hunk 1 above)
Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
| A-DDS-XRCE | Q-MIDDLEWARE | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```

Comment (iluetkeb) [inline articles/embedded.md]: "proved" sounds very formal. Maybe write "demonstrate" instead. Also "seamless" -> "seamlessly".

btw, I tried adding a commit but it didn't work.

Comment (iluetkeb) [inline articles/embedded.md]: "proved" sounds very formal. Maybe write "demonstrate" instead. Also "seamless" -> "seamlessly".

btw, I tried adding a commit but it didn't work.

File: articles/embedded.md (hunk 2 above)
Comment (iluetkeb) [inline articles/embedded.md]:
```suggestion
*   [EU project OFERA](http://ofera.eu/): The EU project OFERA (Open Framework for Embedded Robot Applications) aims at a ROS 2-compatible stack for MCUs in the range of STM32F4 or STM32L1, i.e. with possibly less than 100kB RAM. The project partners currently investigate of using the ROS 2 rmw, rcl and rclcpp layers as-is on the Micro XRCE-DDS implementation of the upcoming DDS-XRCE standard. In parallel, a more modular approach in the style of rosserial is investigated. In the project's use-cases, NuttX is considered as primary choice for the RTOS. Beyond the project page, additional details of the project results can be found at [https://microros.github.io/micro-ROS/](https://microros.github.io/micro-ROS/).
```

Comment (BorjaOuterelo) [reply articles/embedded.md]: Yes, we agreed to use Github provided domain https://microros.github.io/micro-ROS/

**Review by vmayoral — DISMISSED** [review]

File: articles/embedded.md [hunk 4]
```diff
@@ -67,7 +67,8 @@ To answer these questions, the OFERA EU project as well as several others have a
 | A-BUILD-ARDUINO | Q-BUILD | Arduino Build | ROBOTIS has explored building all the libraries using the Arduino IDE, cf. [github.com/ROBOTIS-GIT/OpenCR](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ros2-micrortps). This required some manual changes and thus does not scale, but can get you off the ground. |
 | A-NUTTX-LIBCXX | Q-LANG | C++11/higher support for NuttX | Build libxx from the LLVM project on NuttX, as a pre-requisite to building rclcpp. |
 | A-PERF-RCLCPP-RESOURCE | Q-PERF | Determine resource use of rclcpp | |
-
+| A-DDS-XRCE | Q-COMM | Use of DDS-XRCE standard protocol | This OMG standard defines the protocol used by microcontrollers to publish and subscribe data to a DDS Domain, standard in ROS 2. OFERA and ROBOTIS have demonstrated that it is a suitable protocol to seamlessly communicate with microcontrollers. | [DDS-XRCE](https://www.omg.org/spec/DDS-XRCE/) |
```
Comment (ralph-lange) [inline articles/embedded.md:70]: Is the order DDS XRCE correct? I.e. the standard is named DDS XRCE whereas your implementation is named Micro XRCE-DDS?

File: articles/embedded.md [hunk 5]
```diff
@@ -76,10 +77,21 @@ To answer these questions, the OFERA EU project as well as several others have a
...
+*   [XEL Network by ROBOTIS](https://xelnetwork.readthedocs.io):
+  Product which communicate with ROS 2 (DDS) through DDS-XRCE using Micro XRCE-DDS in the firmware of their CommXel board.
+  This CommXel board manages the rest of the boards conforming the XEL Network and interface them to a ROS 2 space.
+  The CommXel could use Ethernet or UART to communicate using DDS-XRCE.
+
 *   [**freeRTPS**](https://github.com/ros2/freertps): A free, portable, minimalist implementation of the RTPS protocol for microcontrollers such as the STM32F7 developed at the OSRF. FreeRTPS shall allow to run ROS 2 with standard DDS as-is on stronger MCUs. This project has been discontinued in 2016.
 
 *   [**ros2_embedded_nuttx**](https://github.com/ros2/ros2_embedded_nuttx): Early port (in 2014) of ROS 2 alpha for the STM32F4Discovery board and the STM3240G eval board running the RTOS NuttX developed by Víctor Mayoral Vilches and Esteve Fernandez at the OSRF.
 
+*   [Renesas GR-ROSE](http://gadget.renesas.com/ja/event/2018/pm_rose.html):
```
Comment (ralph-lange) [inline articles/embedded.md]: Renesas GR-ROSE in bold, please.

File: articles/embedded.md [hunk 6]
```diff
@@ -76,10 +77,21 @@ To answer these questions, the OFERA EU project as well as several others have a
 
 *   [**ROS 2 library for OpenCR by ROBOTIS**](https://github.com/ROBOTIS-GIT/OpenCR/tree/feature-ros2-micrortps/arduino/opencr_arduino/opencr/libraries/ROS2): Tailored and optimized implementation of the ROS 2 publish/subscribe and clock API for the Micro XRCE-DDS (formerly micro-RTPS) implementation of the upcoming XRCE-DDS middleware standard running on an STM32F7.
 
+*   [XEL Network by ROBOTIS](https://xelnetwork.readthedocs.io):
```
Comment (ralph-lange) [inline articles/embedded.md]: XEL Network by Robotis in bold, please, for consistency.

File: articles/embedded.md (hunk 4 above)
Comment (BorjaOuterelo) [reply articles/embedded.md:70]: Hi @ralph-lange. Yes, that's correct. OMG standard acronym is DDS-XRCE, and eProsima's product name is Micro XRCE-DDS.

**Comment by BorjaOuterelo** [comment]
@iluetkeb I have addressed your observations.

**Review by iluetkeb — DISMISSED** [review]

**Review by ralph-lange — DISMISSED** [review]

File: articles/embedded.md (hunk 5 above)
Comment (BorjaOuterelo) [reply articles/embedded.md]: Done

File: articles/embedded.md (hunk 6 above)
Comment (BorjaOuterelo) [reply articles/embedded.md]: Done

**Comment by BorjaOuterelo** [comment]
@iluetkeb @ralph-lange I pushed latest changes following your recommendations. Can you check and we merge this before the ROS 2 PR?.

**Review by iluetkeb — APPROVED** [review]

**Review by vmayoral — APPROVED** [review]

**Pull request merged**